packages/front-end/components/Features/CompareRuleDiffModal.tsx

```diff
@@ -0,0 +1,230 @@
+import { FeatureInterface, FeatureRule } from "back-end/types/feature";
+import { useState } from "react";
+import ReactDiffViewer from "react-diff-viewer";
+import { FaRedo, FaRegCopy } from "react-icons/fa";
+import { useForm } from "react-hook-form";
+import { useAuth } from "@/services/auth";
+import track from "@/services/track";
+import Modal from "@/components/Modal";
+import SelectField from "@/components/Forms/SelectField";
+import Tooltip from "@/components/Tooltip/Tooltip";
+
+interface Props {
+  isDraft: boolean;
+  isLocked: boolean;
+  canEdit: boolean;
+  envs: string[];
+  feature: FeatureInterface;
+  close: () => void;
+  version: number;
+  setVersion: (version: number) => void;
+  mutate: () => void;
+}
+
+type PartialRule = Omit<FeatureRule, "id">;
+
+export default function CompareRuleDiffModal({
+  isLocked,
+  canEdit,
+  isDraft,
+  envs,
+  feature,
+  close,
+  version,
+  setVersion,
+  mutate,
+}: Props) {
+  const [copyingRules, setCopyingRules] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+  const { apiCall } = useAuth();
+
+  const form = useForm({
+    defaultValues: {
+      sourceEnv: "",
+      targetEnv: "",
+    },
+  });
+
+  const sourceEnv = form.watch("sourceEnv");
+  const targetEnv = form.watch("targetEnv");
+
+  async function handleCopyingRules(
+    targetEnv: string,
+    newRules: FeatureRule[]
+  ) {
+    setCopyingRules(true);
+    setError(null);
+    try {
+      const res = await apiCall<{ version: number }>(
+        `/feature/${feature.id}/${version}/copy-rule-set`,
+        {
+          method: "PUT",
+          body: JSON.stringify({
+            targetEnv,
+            newRules,
+          }),
+        }
+      );
+      track("Clone Feature Rule Set", {
+        sourceEnvironment: sourceEnv,
+        targetEnv,
+        rules: newRules,
+      });
+      await mutate();
+      res.version && setVersion(res.version);
+    } catch (e) {
+      setError(e.message);
+    }
+    setCopyingRules(false);
```

Comment:
Might want to wrap this into a `finally`:
```suggestion
    } finally {
      setCopyingRules(false);
    }
```